### PR TITLE
Outbound validation over xsd/jsd

### PR DESCRIPTION
**Pull request by swedit-majgaonkar-scope**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
 		<fusion-adapter-dedupe.version>0.8</fusion-adapter-dedupe.version>
 		<fusion-shared-kernel.version>0.7</fusion-shared-kernel.version>
 		<spring-boot.version>2.7.0</spring-boot.version>
-		<camel.springboot.version>3.17.0</camel.springboot.version>
+		<camel.springboot.version>3.19.0</camel.springboot.version>
```
Comment (ravi-zadpide) [inline pom.xml:22]: revert this change. as it is related to azure service bus

File: pom.xml [hunk 2]
```diff
@@ -151,6 +151,10 @@
 			<groupId>org.apache.camel.springboot</groupId>
 			<artifactId>camel-azure-eventhubs-starter</artifactId>
 		</dependency>
+		<dependency>
```
Comment (ravi-zadpide) [inline pom.xml:154]: revert this change. as it is related to azure service bus

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/orchestrate/OrchestratorRoute.java [hunk 3]
```diff
@@ -98,7 +98,7 @@ public void configure() {
           .setProperty(ERROR_TARGET_URI, constant(source.getErrorTargetUri()))
           .setProperty(TARGET_HEADER_BLACK_LIST, constant(source.getTargetHeaderBlacklist()))
           .bean(EventDataToMapConverter.class, "updateExchangeHeadersBySource")
-          .bean(EventDataToMapConverter.class, "updateExchangeHeadersToString")
+//          .bean(EventDataToMapConverter.class, "updateExchangeHeadersToString")
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/orchestrate/OrchestratorRoute.java:101]: revert this change. as it is related to azure service bus

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 4]
```diff
@@ -29,47 +33,124 @@
...
+import com.scoperetail.fusion.connect.core.config.Event;
+import com.scoperetail.fusion.connect.core.config.FusionConfig;
+import com.scoperetail.fusion.connect.core.config.Source;
+import lombok.Getter;
 
 @Component
 public class ValidatorRoute extends RouteBuilder {
 
+  @Autowired
+  private FusionConfig fusionConfig;
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:74]: Use constructor injection

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 5]
```diff
@@ -29,47 +33,124 @@
...
+  
+  private final String validateSchema = "direct:validateSchema";
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:76]: rename constant to VALIDATE_SCHEMA_ROUTE

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 6]
```diff
@@ -29,47 +33,124 @@
...
-                    exchange.getProperty(Exchange.EXCEPTION_CAUGHT, Throwable.class);
-                final String validationErrors = getValidationErrors(throwable);
-                exchange.setProperty(IS_VALID_MESSAGE, false);
-                exchange.setProperty(STATUS, PARSE_ERROR);
-                exchange.setProperty(ERRORS, validationErrors);
-                log.error("Exception occurred during schema validation: {}", throwable.getMessage());
+        .setProperty(SCHEMA_VALIDATOR_URI, exchangeProperty(VALIDATOR_URI))
+        .setProperty(SCHEMA_FILE_NAME, constant(PAYLOAD_SCHEMA_NAME))
+        .setProperty(SCHEMA_URI_OPTIONS,
+            exchangeProperty(JSON_VALIDATOR_SCHEMA_URI_OPTION))
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:84]: rename to VALIDATOR_SCHEMA_URI_OPTION

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 7]
```diff
@@ -29,47 +33,124 @@
...
+        .to(validateSchema);
+
+    from("direct:validateInboundPayload")
+        .setProperty(SCHEMA_VALIDATOR_URI, exchangeProperty(INBOUND_VALIDATOR_URI))
+        .setProperty(SCHEMA_FILE_NAME,
+            constant(INBOUND_PAYLOAD_SCHEMA_NAME))
+        .setProperty(SCHEMA_URI_OPTIONS,
+            exchangeProperty(JSON_VALIDATOR_SCHEMA_URI_OPTION))
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:92]: INBOUND_SCHEMA_URI_OPTION

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 8]
```diff
@@ -29,47 +33,124 @@
...
+        .to(validateSchema);
+
+    from("direct:validateOutboundPayload")
+        .setProperty(SCHEMA_VALIDATOR_URI,
+            exchangeProperty(OUTBOUND_VALIDATOR_URI))
+        .setProperty(SCHEMA_FILE_NAME,
+            constant(OUTBOUND_PAYLOAD_SCHEMA_NAME))
+        .setProperty(SCHEMA_URI_OPTIONS,
+            exchangeProperty(XML_VALIDATOR_SCHEMA_URI_OPTION))
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:101]: rename to OUTBOUND_SCHEMA_URI_OPTION

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 9]
```diff
@@ -29,47 +33,124 @@
...
+        .to(validateSchema);
+
+    from(validateSchema).choice().when(exchangeProperty(IS_VALID_MESSAGE)).doTry()
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:104]: remove dotry()

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 10]
```diff
@@ -29,47 +33,124 @@
...
+        .process(new Processor() {
+          @Override
+          public void process(Exchange exchange) throws Exception {
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:107]: move this code to method setSchemaUri()

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 11]
```diff
@@ -29,47 +33,124 @@
...
+                }
               }
-            })
-        .end();
+            }
+          }
+        })
+        .log(DEBUG,
+            "Validating message with the schema provided at path: ${exchangeProperty.validUri}")
+        .toD("${exchangeProperty.validUri}").log(DEBUG, "Message Validated successfully")
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:141]: add doTry()

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java [hunk 12]
```diff
@@ -29,47 +33,124 @@
...
+                final Path payloadSchemaPath =
+                    Paths.get(
+                        fusionConfig.getResourceDirectory(),
+                        source.getName(),
+                        event.getEventType(), 
+                        schemaFileName 
+                        + DOT 
+                        + messageFormat.getExtension());
+                if (Files.exists(payloadSchemaPath)) {
+                  final String newValidatorUri =
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:125]: rename to schemaValidatorUri

File: src/main/java/com/scoperetail/fusion/connect/core/config/FusionConfig.java [hunk 13]
```diff
@@ -52,6 +52,8 @@ public class FusionConfig {
   private Map<String, String> sourceTypes;
```
Comment (ravi-zadpide) [inline src/main/java/com/scoperetail/fusion/connect/core/config/FusionConfig.java:52]: revert file

File: pom.xml (hunk 1 above)
Comment (swedit-majgaonkar-scope) [reply pom.xml]: done this change

File: pom.xml (hunk 2 above)
Comment (swedit-majgaonkar-scope) [reply pom.xml]: done with revert

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/orchestrate/OrchestratorRoute.java (hunk 3 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/orchestrate/OrchestratorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 5 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 9 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 10 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 11 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/config/FusionConfig.java (hunk 13 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/config/FusionConfig.java:52]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 12 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java]: done

File: src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java (hunk 4 above)
Comment (swedit-majgaonkar-scope) [reply src/main/java/com/scoperetail/fusion/connect/core/application/route/validate/ValidatorRoute.java:74]: done

**Review by ravi-zadpide — APPROVED** [review]
